### PR TITLE
http_api (ticdc): check --cert-allowed-cn before add server common name

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve? 
#3627 

### What is changed and how it works?
check `cert-allowed-cn` before add server common nam

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix cli don't work when cli cert's common name was not added in config that use for start cdc server.
```

### CODE REVIEW

File: pkg/security/credential.go [hunk 1]
```diff
@@ -92,6 +92,12 @@ func (s *Credential) getSelfCommonName() (string, error) {
 // AddSelfCommonName add Common Name in certificate that specified by s.CertPath
 // to s.CertAllowedCN
 func (s *Credential) AddSelfCommonName() error {
+	// if CertAllowedCN was not specified, we should not use any common name
+	// otherwise, https requests without a certificate will not be accepted by
+	// cdc server
+	if len(s.CertAllowedCN) == 0 {
```
Comment (liuzix) [inline pkg/security/credential.go:98]: Will it be better if this condition is checked in the caller of `AddSelfCommonName`?

Comment (asddongmen) [reply pkg/security/credential.go:98]: yes.

**Review by overvenus — COMMENTED** [review]
Please add tests for bug-fix PR

**Comment by asddongmen** [comment]
@overvenus @hi-rustin @liuzix PTAL thanks~!

File: cdc/http_status_test.go [hunk 2]
```diff
@@ -159,3 +169,136 @@ func testHandleFailpoint(c *check.C) {
...
+		resp.Body.Close()
+		return nil
+	}, retry.WithMaxTries(retryTime), retry.WithBackoffBaseDelay(50), retry.WithIsRetryableErr(cerrors.IsRetryableError))
+	c.Assert(err, check.IsNil)
+}
+
+//
+func (s *httpStatusSuite) TestServerTLSWithCommonName(c *check.C) {
+	defer testleak.AfterTest(c)
+	addr := "127.0.0.1:8302"
```
Comment (overvenus) [inline cdc/http_status_test.go:242]: Could you use "github.com/tikv/pd/pkg/tempurl"?

**Review by overvenus — APPROVED** [review]

File: pkg/security/util.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
```
Comment (ben1009) [inline pkg/security/util.go]: seems the file for test only, should make it something like util_test.go, kind of scary  to saw a certPem in a file

**Review by ben1009 — APPROVED** [review]

File: pkg/security/util.go (hunk 3 above)
Comment (asddongmen) [reply pkg/security/util.go]: done

**Pull request merged**